components/admin_console/license_settings/trial_banner/trial_banner.tsx

```diff
@@ -0,0 +1,248 @@
+import {format} from 'utils/markdown';
+
+import {Preferences, Unique} from 'utils/constants';
+
+import {GlobalState} from 'types/store';
+import store from 'stores/redux_store.jsx';
+
+interface TrialBannerProps {
+    isDisabled: boolean;
+    gettingTrialError: string | null;
+    requestLicense: () => Promise<void>;
+    gettingTrial: boolean;
+    enterpriseReady: boolean;
+    upgradingPercentage: number;
+    handleUpgrade: () => Promise<void>;
+    upgradeError: string | null;
+    restartError: string | null;
+
+    handleRestart: () => Promise<void>;
+
+    openEEModal: any;
+
+    restarting: boolean;
+}
+
+const TrialBanner: React.FC<TrialBannerProps> = ({
+    isDisabled,
+    gettingTrialError,
+    requestLicense,
+    gettingTrial,
+    enterpriseReady,
+    upgradingPercentage,
+    handleUpgrade,
+    upgradeError,
+    restartError,
+    handleRestart,
+    restarting,
+    openEEModal,
+}: TrialBannerProps) => {
+    let trialButton;
+    let upgradeTermsMessage;
+    let content;
+    let gettingTrialErrorMsg;
+
+    const state = store.getState();
+    const getCategory = makeGetCategory();
+    const preferences = getCategory(state, Preferences.UNIQUE);
+    const restartedAfterUpgradePrefValue = preferences.find((pref: PreferenceType) => pref.name === Unique.REQUEST_TRIAL_AFTER_SERVER_UPGRADE);
+
+    const restartedAfterUpgradePrefs = restartedAfterUpgradePrefValue?.value === 'true';
+    const userId = useSelector((state: GlobalState) => getCurrentUserId(state));
+
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        if (upgradingPercentage === 100) {
+            if (!restarting) {
+                savePrefsRestartedAfterUpgrade();
+                handleRestart();
+            }
+        }
+    }, [upgradingPercentage]);
+
+    useEffect(() => {
+        if (restartedAfterUpgradePrefs) {
+            requestLicense();
+
+            // remove the value from the preferences
+            const category = Preferences.UNIQUE;
+            const name = Unique.REQUEST_TRIAL_AFTER_SERVER_UPGRADE;
+            dispatch(savePreferences(userId, [{category, name, user_id: userId, value: ''}]));
+        }
+    }, [restartedAfterUpgradePrefs]);
+
+    const savePrefsRestartedAfterUpgrade = () => {
+        // save in the preferences that this customer wanted to request trial just after the upgrade
+        const category = Preferences.UNIQUE;
+        const name = Unique.REQUEST_TRIAL_AFTER_SERVER_UPGRADE;
+        dispatch(savePreferences(userId, [{category, name, user_id: userId, value: 'true'}]));
+    };
+
+    const eeModalTerms = (
+        <a
+            role='button'
+            onClick={openEEModal}
+        >
+            <FormattedMarkdownMessage
+                id='admin.license.enterprise.upgrade.eeLicenseLink'
+                defaultMessage='Enterprise Edition License'
+            />
+        </a>
+    );
+    if (enterpriseReady && !restartedAfterUpgradePrefs) {
```

Comment:
Do we also need to check if they have already had a trial license or already have a license?

Reply:
it is checked before even rendering the component. In the license settings page.

Reply:
I missed that, do you have the link to where?

Reply:
you didn't missed it, sorry if I confused you, what I meant was that it was just added in my latest changes but it is not in the trial banner component: https://github.com/mattermost/mattermost-webapp/pull/9454/files#diff-6d72897675f79c66f992edffb30b7fdb4b43dd32487493ee787c5846ac69dde6R394